llvm/test/DebugInfo/CAS/AArch64/multiple_cus.ll

```diff
@@ -0,0 +1,106 @@
+  %0 = load float, ptr %a.addr, align 4, !dbg !35
+  %mul = fmul float %0, 5.000000e+00, !dbg !36
+  ret float %mul, !dbg !37
+}
+
+; Function Attrs: mustprogress noinline nounwind optnone ssp uwtable(sync)
+define noundef float @_Z4bar2f(float noundef %x) #0 !dbg !38 {
+entry:
+  %x.addr = alloca float, align 4
+  store float %x, ptr %x.addr, align 4
+  call void @llvm.dbg.declare(metadata ptr %x.addr, metadata !39, metadata !DIExpression()), !dbg !40
+  %0 = load float, ptr %x.addr, align 4, !dbg !41
+  %mul = fmul float %0, 2.000000e+00, !dbg !42
+  ret float %mul, !dbg !43
+}
+
+attributes #0 = { mustprogress noinline nounwind optnone ssp uwtable(sync) "frame-pointer"="non-leaf" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="apple-m1" "target-features"="+aes,+crc,+dotprod,+fp-armv8,+fp16fml,+fullfp16,+lse,+neon,+ras,+rcpc,+rdm,+sha2,+sha3,+v8.1a,+v8.2a,+v8.3a,+v8.4a,+v8.5a,+v8a,+zcm,+zcz" }
+attributes #1 = { nocallback nofree nosync nounwind speculatable willreturn memory(none) }
+
+!llvm.dbg.cu = !{!0, !2}
+!llvm.ident = !{!4, !4}
+!llvm.module.flags = !{!5, !6, !7, !8, !9, !10}
+
+!0 = distinct !DICompileUnit(language: DW_LANG_C_plus_plus_14, file: !1, producer: "clang version 18.0.0 (git@github.com:apple/llvm-project.git c9c19151210716562c81949ac2e5174e58bc8b64)", isOptimized: false, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, nameTableKind: Apple, sysroot: "/")
+!1 = !DIFile(filename: "/Users/shubham/Development/test109275485/a.cpp", directory: "/Users/shubham/Development/llvm-project-cas/llvm-project")
+!2 = distinct !DICompileUnit(language: DW_LANG_C_plus_plus_14, file: !3, producer: "clang version 18.0.0 (git@github.com:apple/llvm-project.git c9c19151210716562c81949ac2e5174e58bc8b64)", isOptimized: true, runtimeVersion: 0, emissionKind: FullDebug, splitDebugInlining: false, nameTableKind: Apple, sysroot: "/")
+!3 = !DIFile(filename: "/Users/shubham/Development/test109275485/b.cpp", directory: "/Users/shubham/Development/llvm-project-cas/llvm-project")
+!4 = !{!"clang version 18.0.0 (git@github.com:apple/llvm-project.git)"}
+!5 = !{i32 7, !"Dwarf Version", i32 4}
+!6 = !{i32 2, !"Debug Info Version", i32 3}
+!7 = !{i32 1, !"wchar_size", i32 4}
+!8 = !{i32 8, !"PIC Level", i32 2}
+!9 = !{i32 7, !"uwtable", i32 1}
+!10 = !{i32 7, !"frame-pointer", i32 1}
+!11 = distinct !DISubprogram(name: "foo2", linkageName: "_Z4foo2i", scope: !12, file: !12, line: 2, type: !13, scopeLine: 2, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition, unit: !0, retainedNodes: !16)
+!12 = !DIFile(filename: "test109275485/a.cpp", directory: "/Users/shubham/Development")
+!13 = !DISubroutineType(types: !14)
+!14 = !{!15, !15}
+!15 = !DIBasicType(name: "int", size: 32, encoding: DW_ATE_signed)
+!16 = !{}
+!17 = !DILocalVariable(name: "a", arg: 1, scope: !11, file: !12, line: 2, type: !15)
+!18 = !DILocation(line: 2, column: 14, scope: !11)
```

Comment:
Do we need all these different DILocations, or could we collapse them to just a few?

Reply:
(maybe turn off column info or manually reduce it)

Reply:
We only care that there are indeed two line table headers, so technically we can delete all the DILocations, but sure I'll reduce them a little